release-20.2: cli,server: fix the profile file name generation

knz opened the pull request:
Backport 1/1 commits from #55364.
Backport 1/1 commits from #55630

/cc @cockroachdb/release

---

knz committed:
cli,server: fix the profile file name generation

Prior to this patch, the timestamp part in the profile filenames was
using go format `.999` for the millisecond part. However, `.999`
causes the millisecond part to be omitted entirely if they happen to
be zero. The correct format was in fact `.000`.

Release note (bug fix): The file names for generated goroutine, CPU
and memory profiles were sometimes incorrect, resulting in repeated
warnings like `strconv.ParseUint: parsing "txt": invalid syntax` in
log files. This has been corrected.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/55366)
<!-- Reviewable:end -->

knz commented:
woah no I made it worse. Need to tweak this.

knz committed:
profilestore: recognize v20.2-beta timestamps

The previous code was generating timestamps with format `.999` which
caused trailing zeroes to be truncated.

Since filenames with this format may still be present in data
directories, this patch makes the code tolerate them.

Release note: None

knz commented:
ready for review now
